### ISSUE TITLE
multi_scan failing after update

### ISSUE THREAD

**Issue by kaybutter** [lyndsey-ferguson/fastlane-plugin-test_center#34]
Hi,
after updating test_center from 2.4.4 to 3.0.5 multi_scan always fails with the following stack trace:

```
TypeError: [!] no implicit conversion of nil into String
  /usr/local/lib/ruby/gems/2.5.0/gems/plist-3.4.0/lib/plist/parser.rb:66:in `exist?'
  /usr/local/lib/ruby/gems/2.5.0/gems/plist-3.4.0/lib/plist/parser.rb:66:in `initialize'
  /usr/local/lib/ruby/gems/2.5.0/gems/plist-3.4.0/lib/plist/parser.rb:29:in `new'
  /usr/local/lib/ruby/gems/2.5.0/gems/plist-3.4.0/lib/plist/parser.rb:29:in `parse_xml'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-plugin-test_center-3.0.5/lib/fastlane/plugin/test_center/helper/test_collector.rb:22:in `testables'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-plugin-test_center-3.0.5/lib/fastlane/plugin/test_center/helper/correcting_scan_helper.rb:36:in `scan'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-plugin-test_center-3.0.5/lib/fastlane/plugin/test_center/actions/multi_scan.rb:23:in `run'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.84.0/fastlane/lib/fastlane/runner.rb:257:in `block (2 levels) in execute_action'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.84.0/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.84.0/fastlane/lib/fastlane/runner.rb:236:in `block in execute_action'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.84.0/fastlane/lib/fastlane/runner.rb:231:in `chdir'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.84.0/fastlane/lib/fastlane/runner.rb:231:in `execute_action'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.84.0/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.84.0/fastlane/lib/fastlane/fast_file.rb:148:in `method_missing'
  Fastfile:58:in `block (2 levels) in parsing_binding'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.84.0/fastlane/lib/fastlane/lane.rb:33:in `call'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.84.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.84.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.84.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.84.0/fastlane/lib/fastlane/lane_manager.rb:57:in `cruise_lane'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.84.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.84.0/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
  /usr/local/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
  /usr/local/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
  /usr/local/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.84.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'
  /usr/local/lib/ruby/gems/2.5.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.84.0/fastlane/lib/fastlane/commands_generator.rb:337:in `run'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.84.0/fastlane/lib/fastlane/commands_generator.rb:42:in `start'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.84.0/fastlane/lib/fastlane/cli_tools_distributor.rb:98:in `take_off'
  /usr/local/lib/ruby/gems/2.5.0/gems/fastlane-2.84.0/bin/fastlane:20:in `<top (required)>'
  /usr/local/bin/fastlane:23:in `load'
  /usr/local/bin/fastlane:23:in `<top (required)>'
```

Any idea what I might be doing wrong?

**Comment by lyndsey-ferguson** [comment]
Hello @kaybutter, I have an idea, but am not quite sure how you found this defect. Overall, it is failing to find an `xctestrun` file. 

If you are calling `multi_scan` with the `:only_testing` option without the `:xctestrun` option, the code will look for that file in the derived data directory, and it will specifically look for it to be named after the `:scheme` that you provided. This problem _should not_ occur if you gave it the `:xctestrun` option with a valid path.

My guess is that your `xctestrun` file is not named after your Scheme, but something else. I cannot remember off of the top of my head why I made the code look for a `xctestrun` file based off of the Scheme.

Can you look in your derived data directory, `Build/Products` and tell me what `xctestrun` files there are and what your `scheme` is?

Also, can you include in this Issue a snippet of code describing how you are calling `multi_scan`?

**Comment by kaybutter** [comment]
I run my UITests like this:
``` ruby
lane :uitest do
  scan(
    build_for_testing: true,
    scheme: "Mail Release Tests",
    devices: [ENV['DEVICE'] || "iPhone 7"],
    fail_build: true
  )
  multi_scan(
    test_without_building: true,
    scheme: "Mail Release Tests",
    devices: [ENV['DEVICE'] || "iPhone 7"],
    try_count: 5
  )
end
```
The idea was to build it once and then to run the UITests multiple times.

This is my Scanfile:
``` ruby
scheme "Mail"
output_types "junit,html"
output_directory "test-reports"
clean true
devices ["iPhone 7"]
configuration "Test"
fail_build false
skip_build true
```

But that scheme gets overridden for the UITests.

In my `Build/Products` Folder I see the following:
* `Mail Release Tests_iphonesimulator11.2-x86_64.xctestrun`
* `Test-iphonesimulator` Folder with the app

**Comment by lyndsey-ferguson** [comment]
FYI, `multi_scan` will build first if you don't provide `test_without_building` if you want to simplify your lane.

**Comment by kaybutter** [comment]
But will it build again for every repeat?

**Comment by lyndsey-ferguson** [comment]
Can you show me the log too? Perhaps when `multi_scan` calls `scan`, that `scan` is using your scan file? If the log shows that it is using the `Mail` scheme, that would indicate that it is looking for the wrong `xctestrun` file.

**Comment by lyndsey-ferguson** [comment]
No, it builds only once.

**Comment by kaybutter** [comment]
The fastlane log won't give me any details about what calls `multi_scan` makes.

The first call to `scan` succeeds. Then `multi_scan` fails:
```
[15:58:23]: ------------------------
[15:58:23]: --- Step: multi_scan ---
[15:58:23]: ------------------------

+----------------------+----------------------+
| Summary for multi_scan (test_center v3.0.5) |
+----------------------+----------------------+
| try_count            | 5                    |
+----------------------+----------------------+

+----------------------------+------------------------------------------------------------------------------------------+
|                                                     Lane Context                                                      |
+----------------------------+------------------------------------------------------------------------------------------+
| DEFAULT_PLATFORM           | ios                                                                                      |
| PLATFORM_NAME              | ios                                                                                      |
| LANE_NAME                  | ios uitest                                                                               |
| SCAN_DERIVED_DATA_PATH     | /Users/mobileadmin/Library/Developer/Xcode/DerivedData/Mail-gnpuwvjwlahdpfdkbhwzjvxclsgm |
| SCAN_GENERATED_PLIST_FILES | []                                                                                       |
| SCAN_GENERATED_PLIST_FILE  |                                                                                          |
+----------------------------+------------------------------------------------------------------------------------------+
[15:58:23]: no implicit conversion of nil into String

+------+------------------------------+-------------+
|                 fastlane summary                  |
+------+------------------------------+-------------+
| Step | Action                       | Time (in s) |
+------+------------------------------+-------------+
| 1    | Verifying fastlane version   | 0           |
| 2    | default_platform             | 0           |
| 3    | unlock_keychain              | 0           |
| 4    | xcversion                    | 0           |
| 5    | Switch to ios bootstrap lane | 0           |
| 6    | update_googleadmob           | 1           |
| 7    | scan                         | 118         |
| 💥   | multi_scan                   | 0           |
+------+------------------------------+-------------+

[15:58:23]: fastlane finished with errors
```

**Comment by lyndsey-ferguson** [comment]
I think this path, `/Users/mobileadmin/Library/Developer/Xcode/DerivedData/Mail-gnpuwvjwlahdpfdkbhwzjvxclsgm` confirms it for me. `scan` created that path, but I believe that `multi_scan` will look for the `Scheme` given to it.

**Comment by kaybutter** [comment]
So, removing my own call to `scan` and having `multi_scan` take care of the building seems to fix this particular problem. But then I run into problems with the `clean: true` in my `Scanfile`. :(

`multi_scan` seems to build the project successfully, then it cleans the result before trying to run the tests which then fails. If I pass `clean: false` to `multi_scan` it will pass it to the first build step but not to the following test runs which will just use the options from the Scanfile :(

So the only solution for now seems to be removing the `clean: true` from the `Scanfile`...

**Comment by lyndsey-ferguson** [comment]
Thank you for further investigating this. It seems that the `Scanfile's` options override whatever we send (or do not send?) to the actual `scan` command.

I'm glad you have a workaround for now, this issue will be the next one I tackle after #31

**Comment by lyndsey-ferguson** [comment]
Fixed in [v3.0.8](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/tree/v3.0.8)
